objective.py

```diff
@@ -56,17 +56,17 @@ def compute(
             c2st_mean, c2st_std = None, None
             emd_mean, emd_std = None, None
             mmd_mean, mmd_std = None, None
+            sampling_time = None
         else:
             start = time.perf_counter()
-            theta_est = [
-                sample(x, self.theta_ref[i].shape[0])
-                for i, x in enumerate(self.x_ref)
-            ]
+            n_samples = self.theta_ref[0].shape[0]
+            theta_est = [sample(x, n_samples) for x in self.x_ref]
```

Comment:
I think this could be problematic if all reference posteriors don't have the same size.

```suggestion
            n_ref = (theta.shape[0] for theta in self.theta_ref)
            theta_est = [sample(x, n) for x, n in zip(self.x_ref, n_ref)]
```